app/src/main/java/com/example/superimagesviewer/MosaicRepository.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.superimagesviewer
+
+import android.app.Application
+import androidx.lifecycle.MutableLiveData
+import android.graphics.drawable.Drawable
+import androidx.lifecycle.LiveData
+import android.os.AsyncTask
+import java.io.IOException
+import java.lang.ref.WeakReference
+
+class MosaicRepository(application: Application) {
+    val mosaicsList = MutableLiveData<List<Drawable>>()
+    private val drawables = mutableListOf<Drawable>()
+    fun getMosaicsList(): LiveData<List<Drawable>> {
+        return mosaicsList
+    }
+
+    private inner class InsertAsyncTask(application: Application) : AsyncTask<Void, Void, Void>() {
```

Comment:
Please avoid usage of AsyncTask, use kotlin coroutine instead

Reply:
done